.github/workflows/appimage.yml

```diff
@@ -28,21 +34,36 @@ jobs:
       options: --platform ${{ matrix.platform }}
     timeout-minutes: 15
     outputs:
-      date: ${{ steps.build_test_app.outputs.date }}
+      date: ${{ steps.build_date.outputs.date }}
 
     steps:
-      - name: Check out main branch code
+      - name: Check out code
+        if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
```

Comment:
[note] This additional if is used in tandem with requests for PR appimages which trigger on 'pull_request_target' however it can be removed if desired

Reply:
This doesn't make sense to have here because none of the later steps will work without code checked out first, so this isn't a "sometimes" step as far as I can tell, unless you've added a different checkout step for different conditions, but I don't see one.